spreadsheet_oca/controllers/main.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 Tecnativa - Carlos Roca
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import json
+from io import BytesIO
+from zipfile import ZipFile
+
+from odoo.http import Controller, content_disposition, request, route
+
+
+class SpreadsheetDownloadXLSX(Controller):
+    @route("/spreadsheet/xlsx", type="http", auth="user", methods=["POST"])
+    def download_spreadsheet_xlsx(self, zip_name, files, **kw):
+        files = json.loads(files)
```

Comment:
Crashes when exporting to XLSX when filestorage object not JSON.

Add this line above:
`if hasattr(files, "read"):
    files = files.read().decode("utf-8")`